File: prime_number_primer/prime_number_primer__eratosthenes.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        bool[] p = Repeat( true, n + 1 ).ToArray();

        p[0] = false;
        p[1] = false;
        for( int i = 2; i <= n; i++ ) {
            if( p[i] == true ) {
                for( int j = 2 * i; j <= n; j += i ) {
                    p[j] = false;
                }
            }
        }
        WriteLine( p[n] ? "YES" : "NO" );
        return;
    }
}

File: prime_number_primer/prime_number_primer__eratosthenes.py
# 実行時間 120ms
n = int(input())
p = [True] * (n + 1)

p[0] = False
p[1] = False
for i in range(2, n + 1):
    if p[i] == True:
        j = 2 * i
        while j <= n:
            p[j] = False
            j += i
print("YES" if p[n] else "NO")
